docs/app/components/component-section/component-section.component.ts

```diff
@@ -31,27 +36,58 @@ export class ComponentSectionComponent implements OnInit {
     deprecatedLink: ILink;
     hybridModuleTs: string = require('!!raw-loader!./snippets/hybrid-module.ts');
 
+    private _documentationSection: T;
+    private _onDestroy = new Subject<void>();
+
     constructor(
         private _resolverService: ResolverService,
-        private _navigationService: NavigationService
+        private _navigationService: NavigationService,
+        private _siteThemeService: SiteThemeService
     ) { }
 
     ngOnInit(): void {
         const component = documentationSectionNames[this.componentName];
 
-        if (component) {
-            const factory = this._resolverService.resolveComponentFactory(component);
-            const componentRef = this.viewContainer.createComponent(factory);
-
-            if (isIPlaygroundProvider(componentRef.instance)) {
-                this.playground = (<IPlaygroundProvider>componentRef.instance).playground;
-            }
-        } else {
+        if (!component) {
             console.warn(`ComponentSectionComponent: ${this.componentName} cannot be resolved - decorate component with @DocumentationSectionComponent.`);
+            return;
         }
 
+        const factory = this._resolverService.resolveComponentFactory(component);
+        const componentRef = this.viewContainer.createComponent<T>(factory);
+        this._documentationSection = componentRef.instance;
+
+        this.updateWithTheme(this._siteThemeService.theme$.getValue());
```

Comment:
I'm not sure if `updateWithTheme` clearly describes the purpose of this. From what I can see so far, it looks like it switches the theme value in `BaseDocumentationSection` and updates the snippets. If in future I were to arrive in this file and read this line of code, the function name doesn't really indicate its purpose and I'll have to read the actual content of the function. Ideally we should be able to clearly understand the purpose of a function just from its name. Could `updateWithTheme` be renamed to possibly better reflect that it's switching to the theme specific snippets?

Reply:
It's a function which is called when the theme changes, I'm sure sure what else to call it. `onThemeChange`?

Reply:
I'm not sure `onThemeChange` would work either. Since from the perspective of `BaseDocumentationSection` it's actually updating the theme. I think something along the lines of `useSnippetsForTheme(theme)` or `replaceSnippetsForTheme(theme)`. With those names (or something similar you might be able to think of), you have the advantage of being able to know what the purpose of the function is without having to open the function and read/understand its contents. Currently, `updateWithTheme` gives no indication that the snippets are being replaced with their theme specific alternatives.

Reply:
Compare it with ngOnInit for example. If we need to make it do something in addition to changing the snippets, `useSnippetsForTheme` would become misleading.

Reply:
If we were doing something else other than just updating the snippets then I agree `onThemeChange` would be fine. But since currently all we're doing is replacing the contents of the snippets then I think we could take advantage of the simplicity and readability of having a function named something like `useSnippetsForTheme` (or if you can think of anything better). If we need to add more to that function in the future then we could always rename it to better reflect its purpose at that point.

Reply:
Right now it does do multiple things:
```ts
    private updateWithTheme(theme: SiteThemeId): void {
        // Some sections without snippets don't extend BaseDocumentationSection, ignore those
        if (isBaseDocumentationSection(this._documentationSection)) {
            this._documentationSection.updateWithTheme(theme);
            this.updatePlayground();
        }
    }
```

Reply:
My bad, I probably could've put this comment on a different line to be more clear. I meant the `updateWithTheme` function in the `BaseDocumentationSection` class which is called on line 79 of this file. Although now that I notice it, it probably isn't good to have two functions both called `updateWithTheme`, with one of them calling the other.

The private function in this component makes sense named as `updateWithTheme` (possibly without the 'with'). A function which is updating a theme in this scenario seems to have two actions:

1. Updating the snippets (via line 79, which with the current naming doesn't indicate that this is happening)
1. Updating the playground

By renaming the `BaseDocumentationSection`'s `updateWithTheme` function, then when you read the private function in this class you'd be able to see these two actions without having to open and read through the functions.

Reply:
Thanks, yes they should not share the same name, this is a good point.

The function in `BaseDocumentationSection` also does two separate things, and if a documentation section overrides it to hook into the event, possibly more.

I've done some renaming which should hopefully make the purpose clear.

Reply:
Thanks, I'm happy with the updated naming. Definitely seems clearer and easier to follow now 👍

Reply:
Great, thank you for the feedback!